src/neuroconv/tools/data_transfers/_aws.py

```diff
@@ -30,3 +36,289 @@ def estimate_s3_conversion_cost(
+        service_name="dynamodb",
+        region_name=region,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+    dynamodb_resource = boto3.resource(
+        service_name="dynamodb",
+        region_name=region,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+    iam_client = boto3.client(
+        service_name="iam",
+        region_name=region,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+    batch_client = boto3.client(
+        service_name="batch",
+        region_name=region,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+
+    # It is very useful to have a status tracker that is separate from the job environment
+    # Technically detailed logs of inner workings are given in the CloudWatch, but that can only really be
+    # analyzed from the AWS web console
+    current_tables = dynamodb_client.list_tables()["TableNames"]
+    if status_tracker_table_name not in current_tables:
+        table = dynamodb_resource.create_table(
+            TableName=status_tracker_table_name,
+            KeySchema=[dict(AttributeName="id", KeyType="HASH")],
+            AttributeDefinitions=[dict(AttributeName="id", AttributeType="S")],
+            ProvisionedThroughput=dict(ReadCapacityUnits=1, WriteCapacityUnits=1),
+        )
+    else:
+        table = dynamodb_resource.Table(name=status_tracker_table_name)
+
+    # Ensure role policy is set
+    current_roles = [role["RoleName"] for role in iam_client.list_roles()["Roles"]]
+    if iam_role_name not in current_roles:
+        assume_role_policy = dict(
+            Version="2012-10-17",
+            Statement=[
+                dict(Effect="Allow", Principal=dict(Service="ecs-tasks.amazonaws.com"), Action="sts:AssumeRole"),
+            ],
+        )
+
+        role = iam_client.create_role(RoleName=iam_role_name, AssumeRolePolicyDocument=json.dumps(assume_role_policy))
+        iam_client.attach_role_policy(
+            RoleName=role["Role"]["RoleName"], PolicyArn="arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess"
+        )
+        iam_client.attach_role_policy(
+            RoleName=role["Role"]["RoleName"], PolicyArn="arn:aws:iam::aws:policy/CloudWatchFullAccess"
+        )
+    else:
+        role = iam_client.get_role(RoleName=iam_role_name)
+
+    # Ensure compute environment is setup
+    current_compute_environments = [
+        environment["computeEnvironmentName"]
+        for environment in batch_client.describe_compute_environments()["computeEnvironments"]
+    ]
+    if compute_environment_name not in current_compute_environments:
+        batch_client.create_compute_environment(
+            computeEnvironmentName=compute_environment_name,
+            type="MANAGED",
+            state="ENABLED",
+            computeResources={
```

Comment:
TODO: enable spot and expose % as keyword argument

Reply:
default should be `100%` for minimum delay, but some users (such as us for big non-priority data churning) might want to manually set it lower to save costs

Reply:
Actually after looking more into it, I recall that we don't intend to immediately support this because the processes can be easily interrupted by anyone else paying for on-demand, and these are such long processes usually that the risk of losing the instance (and thus any money spent to that point) is too high